lib/backend/k8s/conversion.go

```diff
@@ -337,6 +354,8 @@ func (c converter) k8sIngressRuleToCalico(r extensions.NetworkPolicyIngressRule,
 	}
 
 	// Combine desintations with sources to generate rules.
+	// TODO: This currently creates a lot of rules by making every combination of from / ports
+	// into a rule.  We can combine these so that we don't need as many rules!
```

Comment:
There are two possible compactions here.
- Instead of one port per rule, you should be able to do one protocol per rule.
- You could just ()||() all the selectors together (although we don't have a data model reference for anything >1.4; I'm assuming we haven't modified the allowed syntax).